pkg/lib/v0_2_0/events.go

```diff
@@ -65,14 +66,25 @@ func NewHTTPEventSender(endpoint string) (*HTTPEventSender, error) {
 	return httpSender, nil
 }
 
-// SendEvent sends a CloudEvent
+// Deprecated: use HTTPEventSender.Send instead
 func (httpSender HTTPEventSender) SendEvent(event cloudevents.Event) error {
 	ctx := cloudevents.ContextWithTarget(context.Background(), httpSender.EventsEndpoint)
 	ctx = cloudevents.WithEncodingStructured(ctx)
 	return httpSender.Send(ctx, event)
 }
 
 func (httpSender HTTPEventSender) Send(ctx context.Context, event cloudevents.Event) error {
+
+	// if not already added by the caller, add the one from the Sender
+	if cloudevents.TargetFromContext(ctx) == nil {
+		ctx = cloudevents.ContextWithTarget(ctx, httpSender.EventsEndpoint)
+	}
+
+	// TODO: Should we also add cloudevents.WithEncodingStructured(ctx) here to avoid duplication?
```

Comment:
@bacherfl @warber what you think about this? Should we always set `cloudevents.WithEncodingStructured(ctx)` in here, as it was done before in `SendEvent`? This will make transitioning to this version easier. 

I get that by doing so we lose flexibility in case we want to use binary encoding later. But is that even the case? I tried checking if such encoding config is already present, but as my comment says the key added to `ctx` is not exported in `cloudevents`.